sessionctx/binloginfo/binloginfo.go

```diff
@@ -49,14 +51,75 @@ type BinlogInfo struct {
 	Client *pumpcli.PumpsClient
 }
 
-// GetPumpsClient gets the pumps client instance.
+// GetPumpsClient get the pumps client instance.
 func GetPumpsClient() *pumpcli.PumpsClient {
 	pumpsClientLock.RLock()
 	client := pumpsClient
 	pumpsClientLock.RUnlock()
 	return client
 }
 
+// GetPumpsClientByLogBin get the pumps client instance by log_bin variable.
+func GetPumpsClientByLogBin() *pumpcli.PumpsClient {
+	var binlogCli *pumpcli.PumpsClient
+	tidbLogBin := variable.GetSysVar(variable.TiDBLogBin)
+	logBin := variable.GetSysVar(variable.LogBin)
+	if (logBin.Scope&variable.ScopeSession == 1 && logBin.Value == "1") ||
+		(logBin.Value == "auto" && tidbLogBin.Scope&variable.ScopeSession == 1 && tidbLogBin.Value == "1") {
+		atomic.StoreUint32(&variable.ProcessGeneralLog, 1)
+		binlogCli = GetOrCreatePumpsClient()
+	}
+	return binlogCli
+}
+
+// GetOrCreatePumpsClient get or create the pumps client instance
+func GetOrCreatePumpsClient() *pumpcli.PumpsClient {
+	client := GetPumpsClient()
+	if client == nil {
```

Comment:
you can take a look 

https://github.com/pingcap/tidb-tools/blob/a17aa9c690a7a055a5e47d1c8295ca5f55f46ef3/tidb-binlog/pump_client/pump.go#L135